it/uk/gov/hmrc/excisemovementcontrolsystemapi/DraftExciseMovementControllerItSpec.scala

```diff
@@ -275,7 +275,7 @@ class DraftExciseMovementControllerItSpec extends PlaySpec
   private def assertValidResult(result: WSResponse) = {
     val responseBody = Json.parse(result.body).as[ExciseMovementResponse]
     responseBody.status mustBe "Accepted"
-    responseBody.movementId.value.toString must fullyMatch.regex("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
+    responseBody.movementId must fullyMatch.regex("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
```

Comment:
I am not sure If I would go that far to test the Id (e.g. use the match with reg expression )as this is generated by the UUID library, so we can be sure that it will generated one of this type. If you really want to make sure that the ID is not directly passed as a string to the Movement object you may want to changes the access level to private for the case class Movement, so the design will enforce that, as we only be able to use the apply function which uses the UUID.randomUUID() function.

So the test will be easier (rex expression are difficult to read ) as you may check that the id is not empty

Reply:
Could do and I agree with the concern that we are just testing the UUID library,  but Mike had reasoning for suggesting a regex in the comments here: https://github.com/hmrc/excise-movement-control-system-api/pull/65

Reply:
my be better enforce this by design as I explained above, so there will be no danger/chance to create an ID that is not a UUID (this would be better in my opinion), but you could also use UUID.fromString(movementId) in the test as this will throw and the test fails if is not the right format. This may be more readable then using a regular expression